packages/backend/src/app/flows/flow-version/flow-version.service.ts

```diff
@@ -288,47 +325,53 @@ async function prepareRequest(projectId: ProjectId, flowVersion: FlowVersion, re
+                    break
+                }
             }
             break
         case FlowOperationType.UPDATE_ACTION:
             clonedRequest.request.valid = true
-            if (clonedRequest.request.type === ActionType.MISSING) {
-                clonedRequest.request.valid = false
-            }
-            else if (clonedRequest.request.type === ActionType.LOOP_ON_ITEMS) {
-                clonedRequest.request.valid = loopSettingsValidator.Check(clonedRequest.request.settings)
-            }
-            else if (clonedRequest.request.type === ActionType.BRANCH) {
-                clonedRequest.request.valid = branchSettingsValidator.Check(clonedRequest.request.settings)
-            }
-            else if (clonedRequest.request.type === ActionType.PIECE) {
-                clonedRequest.request.valid = await validateAction(clonedRequest.request.settings)
-            }
-            else if (clonedRequest.request.type === ActionType.CODE) {
-                const codeSettings: CodeActionSettings = clonedRequest.request.settings
-                await uploadArtifact(projectId, codeSettings)
-                const previousStep = flowHelper.getStep(flowVersion, clonedRequest.request.name)
-                if (
-                    previousStep !== undefined &&
-                    previousStep.type === ActionType.CODE &&
-                    codeSettings.artifactSourceId !== previousStep.settings.artifactSourceId
-                ) {
-                    await deleteArtifact(projectId, previousStep.settings)
+            switch (clonedRequest.request.type) {
+                case ActionType.MISSING:
+                    clonedRequest.request.valid = false
+                    break
+                case ActionType.LOOP_ON_ITEMS:
+                    clonedRequest.request.valid = loopSettingsValidator.Check(clonedRequest.request.settings)
+                    break
+                case ActionType.BRANCH:
+                    clonedRequest.request.valid = branchSettingsValidator.Check(clonedRequest.request.settings)
+                    break
+                case ActionType.PIECE:
+                    clonedRequest.request.valid = await validateAction(clonedRequest.request.settings)
+                    break
+                case ActionType.CODE: {
+                    const codeSettings: CodeActionSettings = clonedRequest.request.settings
+                    await uploadArtifact(projectId, codeSettings)
+                    const previousStep = flowHelper.getStep(flowVersion, clonedRequest.request.name)
+                    if (
+                        previousStep !== undefined &&
+                        previousStep.type === ActionType.CODE &&
+                        codeSettings.artifactSourceId !== previousStep.settings.artifactSourceId
+                    ) {
+                        await deleteArtifact(projectId, previousStep.settings)
+                    }
+                    break
```

Comment:
The `prepareRequest` method is gigantic, I think it will be better to split it to multiple methods or even files

Reply:
I will do this in follow up PR.